lib/ace/mode/latex_parser.js

```diff
@@ -1,30 +1,59 @@
 define(function(require, exports, module) {
 "use strict";
 
-var WORD_PATTERN = "[a-zA-Zа-яА-ЯёЁ]+";
-var WORD_REGEX = new RegExp("^" + WORD_PATTERN);
+var WORD_REGEX = /^[a-zа-яё]+/i,
+    // % * <foo@bar.com> 2015-01-12T00:17:27.829Z:
+    DISCUSSION_COMMENT_REGEXP = /^\s*%\s([\*\^])\s<(.+)>(\s(.+))?:$/,
+    DISCUSSION_TEXT_ROW_REGEXP = /^\s*%\s*(.*)/;
 
 var LatexParser = exports.LatexParser = function(bgTokenizer) {
     this.myErrors = [];
+    this.myDiscussions = [];
     this.myTokenizer = bgTokenizer;
 };
 
 (function(){
+    this.setSpellingCheckDictionary = function(dictionary) {
+        this.myDictionary = dictionary;
+    };
 
-    this.go = function(doc, dictionary){
-        if (dictionary == null) {
+    /**
+     * Parse the document
+     * @param doc              document object
+     * @param doSpellcheck     true if we want spellcheck
+     * @param doGetDiscussions true if we want get text comments
+     */
+    this.go = function(doc, doSpellcheck, doGetDiscussions) {
```

Comment:
If we replace the last two params with object with properties like `spellcheck` and `parseDiscussions` then it will be more clear in client code: 

    go(doc, {
      spellcheck: true
      parseComments: false
    }

rather than `go(doc, true, false`

What do you think about it?